apcd-cms/src/client/src/components/Extensions/EditExtensionModal/EditExtensionModal.tsx

```diff
@@ -0,0 +1,380 @@
+                    </Label>
+                    <Field
+                      as="select"
+                      name="ext_outcome"
+                      id="ext_outcome"
+                      onChange={formik.handleChange}
+                      onBlur={formik.handleBlur}
+                      className={`form-control ${styles.viewRecord}`}
+                      value={formik.values.ext_outcome}
+                    >
+                      {outcomeOptions?.map((opt) => (
+                        <option
+                          className="dropdown-text"
+                          key={opt.key}
+                          value={opt.value}
+                        >
+                          {opt.value}
+                        </option>
+                      ))}
+                    </Field>
+                    <ErrorMessage
+                      name="ext_outcome"
+                      component="div"
+                      className="text-danger"
+                    />
+                  </FormGroup>
+                </Col>
+                <Col md={6}>
+                  <FormGroup>
+                    <Label for="notes" className={styles.customLabel}>
+                      <strong>Notes</strong>
+                    </Label>
+                    <Field
+                      as="textarea"
+                      name="notes"
+                      id="notes"
+                      rows="5"
+                      maxLength="2000" // Set the maxLength attribute
+                      onChange={formik.handleChange}
+                      onBlur={formik.handleBlur}
+                      className={`form-control ${styles.viewRecord}`}
+                    />
+                    <small
+                      className="form-text text-muted"
+                      style={{ fontStyle: 'italic' }}
+                    >
+                      2000 character limit
+                    </small>
+                    <ErrorMessage
+                      name="notes"
+                      component="div"
+                      className="text-danger"
+                    />
+                  </FormGroup>
+                </Col>
+              </Row>
+              <br />
+              <Alert color="danger" isOpen={showErrorMessage}>
+                Error: {errorMessage}
+              </Alert>
+              <Button
+                type="submit"
+                color="primary"
+                disabled={formik.isSubmitting}
+                className={styles.customSubmitButton}
+              >
+                Submit
+              </Button>
+            </form>
+          </FormikProvider>
+          <hr />
+          <div className={styles.viewRecord}>
+            {userFields.map((field, index) => (
+              <Row key={index}>
+                <Col md={6}>
+                  <p>{field.label}:</p>
+                </Col>
+                <Col md={6}>
+                  <p>{field.value}</p>
```

Comment:
The updated field values are not refreshing here after a submit on the form to edit but they are populating in the form correctly. Maybe a use effect could go here to pickup changes?

Reply:
`userFields` is not bound to the data in the form.
To allow binding, react provide `state` and/or `useEffect` features.

1. Set state
```
const [userFields, setUserFields] = useState([
    { label: 'Applicable Data Period', value: extension?.applicable_data_period || 'None' },
    { label: 'Approved Expiration Date', value: extension?.approved_expiration_date || 'None' },
    { label: 'Exception Status', value: extension?.ext_status },
    { label: 'Exception Outcome', value: extension?.ext_outcome },
    { label: 'Exception Notes', value: extension?.notes || 'None' },
  ]);
```
2. in onSubmit, within this condition
if (onEditSuccess && response) {
        onEditSuccess(response);
      }
add below code
```
setUserFields([
          { label: 'Applicable Data Period', value: values.applicable_data_period || 'None' },
          { label: 'Approved Expiration Date', value: values.approved_expiration_date || 'None' },
          { label: 'Exception Status', value: values.ext_status },
          { label: 'Exception Outcome', value: values.ext_outcome },
          { label: 'Exception Notes', value: values.notes || 'None' },
        ]);
 ```